lintro/ai/review/chunker.py

```
@@ -0,0 +1,975 @@
+        cleaned = cleaned[2:]
+    base = _normalize_posix_shell_path(path=cwd)
+    if base:
+        return _normalize_posix_shell_path(path=f"{base}/{cleaned}")
+    return _normalize_posix_shell_path(path=cleaned)
+
+
+def _shell_cwd_after_line(*, line: str, cwd: str) -> str:
+    """Return cwd after processing ``cd`` commands on a shell line."""
+    current = _normalize_posix_shell_path(path=cwd)
+    for segment in re.split(r"&&|\|\||;|\|", line):
+        segment = segment.strip()
```

Ruff (0.15.18)

[warning] `for` loop variable `segment` overwritten by assignment target

(PLW2901)